sound/soc/sof/ipc.c

```diff
@@ -268,7 +268,6 @@ static int sof_ipc_tx_message_unlocked(struct snd_sof_ipc *ipc, u32 header,
 	/* initialise the message */
 	msg = &ipc->msg;
 
-	msg->header = header;
```

Comment:
@RanderWang is this change compatible with IPC4?

Reply:
it will affect IPC4 but I will not use header if this PR is accepted

Reply:
In my view the snd_sof_ipc_msg.header is redundant as it is the same as the hdr.cmd.
The only reason which can justify is to use the sof_ipc_tx_message/sof_ipc_tx_message_no_pm to _not_ send a message (msg_data = NULL, msg_size = 0) which does not make much sense as per API.

The sof_ipc_tx_message/sof_ipc_tx_message_no_pm parameter list can be simplified as well to:
`(struct snd_sof_ipc *ipc, void *ipc_msg, void *reply_data, size_t reply_bytes)`
and internally use the header's cmd and size.

Yes, it is going to be a bit of churn, but it will make the API a bit simpler to use, after all _all_ users of this must prepare the hdr.cmd and hdr.size before giving it for sending.

Reply:
header is duplicated in the body so it can be dropped from API. The API needs to be ABI generic too so passing a buffer and size is generic and correct.

However, Intel HW does support "compact messages" i.e. where message fits into 2 words and this is where having a ```int compact``` flag in the API makes sense as it tell the HW (dont use mailbox, write the buffer contents using your compact message registers).

Reply:
if the message only contains the hdr (2x u32) then the hdr.size = 8 and in this case the code can decide that we can use compact message mode for this?

Reply:
Lets have a flag so that clients can set/clear it depending on IPC major and use case.